app/src/main/java/uk/gov/onelogin/developer/tabs/app/AppTabScreenViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.onelogin.developer.tabs.app
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import uk.gov.onelogin.appinfo.apicall.domain.model.AppInfoData
+import uk.gov.onelogin.appinfo.source.domain.model.AppInfoLocalState
+import uk.gov.onelogin.appinfo.source.domain.source.AppInfoLocalSource
+
+@HiltViewModel
+class AppTabScreenViewModel @Inject constructor(
+    private val appInfoLocalSource: AppInfoLocalSource
+) : ViewModel() {
+    @OptIn(ExperimentalSerializationApi::class)
+    fun getAppInfoAsString(): String {
+        return when (val appInfoStatus = appInfoLocalSource.get()) {
+            is AppInfoLocalState.Failure ->
+                "Failed to retrieve app info"
+
+            is AppInfoLocalState.Success -> {
```

Comment:
Isn't the whole point of using a remote or local source abstraction to give us back something usable like a data class, hiding away any serialisation or deserialisation form the UI layers?

Reply:
To some extent yes, for general use in our application then yes. Here we are just wanting to print the whole lot to the screen in a simple way for the dev menu viewing